lib/internal/options.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
```

Comment:
I have a few questions:
1. Why a new module (this also had bootstrap overhead)>
2. Why internal, IMO this is information that's useful for the user (similar to `process.execArgv`, `process.debugPort`, `process.config`, etc.)?

Also maybe the name should be more like `cliOptions` or `runtimeOptions`?

Reply:
1. The overhead is negligible (my local benchmark shows a low-confidence 0.04% improvement....)
2. It was internal, this is just caching results from internal bindings. I believe we will have to eventually implement process.config with this when we remove node_config.cc? @addaleax

There can be another follow up PR to rename the properties, sure. This PR is just caching the existing properties.

Reply:
> 1. The overhead is negligible

I'm +0.1 on your approach, I this good encapsulation & better maintainability is more important than negligible overhead. Just GYI there are others who disagree (that's one of the reasons `test/parallel/test-bootstrap-modules.js` exists 


> 2. It was internal, this is just caching results from internal bindings.

Ack.

Reply:
If we provide some user-facing way to access options (which I’d like), I think we’d want to put some thought into the API design for that. Having something that we use internally and iterate on seems like a good way forward.

(`process.config` is a different thing, I’m just assuming we’re talking about `process.binding('config')`)

Reply:
I was thinking something as simple as `process.config` is, i.e. an immutable json like object.

But I'm 👍 on proper API design.